mapit_mysociety_org/settings.py

```diff
@@ -3,11 +3,46 @@
 
 # Import MapIt's settings (first time to quiet flake8)
 from mapit_settings import (
-    config, INSTALLED_APPS, TEMPLATES, MIDDLEWARE, STATICFILES_DIRS, BASE_DIR, MAPIT_RATE_LIMIT, PARENT_DIR)
+    config, INSTALLED_APPS, TEMPLATES, MIDDLEWARE, STATICFILES_DIRS, BASE_DIR, MAPIT_RATE_LIMIT, PARENT_DIR, DATABASES)
 from mapit_settings import *  # noqa
 
 # Update a couple of things to suit our changes
 
+if config.get('MAPIT_DB_RO_HOST', '') and 'test' not in sys.argv:
+    DATABASES['replica'] = {
+        'ENGINE': 'django.contrib.gis.db.backends.postgis',
+        'NAME': config.get('MAPIT_DB_NAME', 'mapit'),
+        'USER': config.get('MAPIT_DB_USER', 'mapit'),
+        'PASSWORD': config.get('MAPIT_DB_PASS', ''),
+        'HOST': config['MAPIT_DB_RO_HOST'],
+        'PORT': config['MAPIT_DB_RO_PORT'],
+        # Does not appear to work
+        # 'TEST': {
+        #     'MIRROR': 'default',
+        # },
+    }
+
+    import random
+    from .multidb import use_primary
+
+    class PrimaryReplicaRouter(object):
+        def db_for_read(self, model, **hints):
+            if use_primary():
+                return 'default'
+            return random.choice(['default', 'replica'])
+
+        def db_for_write(self, model, **hints):
+            return 'default'
+
+        def allow_relation(self, obj1, obj2, **hints):
+            return True
+
+        def allow_migrate(self, db, app_label, model_name=None, **hints):
+            return True
```

Comment:
The docs say:

> Determine if the migration operation is allowed to run on the database with alias `db`.

Presumably running migrations against the `replica` DB will fail because it's a read-only standby, so should `False` be returned in this case?

Reply:
`migrate` only ever runs against one database, which is default unless explicitly given, so this will never be run against replica.